File: ios/snacks/src/OArray.swift
//
//  OArray.swift
//  snacks
//
//  Created by 谢月甲 on 16/2/21.
//  Copyright © 2016年 noear. All rights reserved.
//

import Foundation

public class OArray{
    public final var elements:[ONode] = [];
    
    public func add(value:ONode){
        elements.append(value);
    }
    
    public func add(index:Int,_ value:ONode){
        elements.insert(value, atIndex: index);
    }
    
    public func get(index:Int) -> ONode{
        return elements[index];
    }
    
    public func count()->Int{
        return elements.count;
    }
    
    public func clear(){
        elements.removeAll();
    }
}

File: android/Snacks/src/main/java/noear/snacks/OArray.java
package noear.snacks;

import noear.snacks.ONode;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by noear on 14-6-11.
 */
public class OArray {
    public List<ONode> elements;

    public  OArray(){
         elements = new ArrayList<ONode>();
    }

    public void add(ONode value)
    {
        elements.add(value);
    }

    public void add(int index, ONode value)
    {
        elements.add(index,value);
    }

    public ONode get(int index)
    {
        return elements.get(index);
    }

    public int count(){
        return elements.size();
    }

    public void clear(){
        elements.clear();
    }
}
